### ISSUE TITLE
Unable to resolve reference to template class or its member inside a namespace

### ISSUE THREAD

**Issue by lahwaacz** [doxygen/doxygen#9678]
Consider this dummy C++ code:
```cpp
namespace foo {

/**
 * \brief Some base class
 */
template< typename T >
class Base
{
public:
   /**
    * \brief Alias for the template parameter of \ref Base
    */
   using ValueType = T;

   /**
    * \brief Some member function
    */
   void member_1();

   /**
    * \brief Some member function like \ref member_1
    */
   void mmeber_2();
};

class Model : public Base<int> {};

}  // namespace foo
```
Running doxygen gives these errors:
```
/home/lahwaacz/stuff/test.cpp:21: warning: unable to resolve reference to 'member_1' for \ref command
/home/lahwaacz/stuff/test.cpp:11: warning: unable to resolve reference to 'Base' for \ref command
/home/lahwaacz/stuff/test.cpp:11: warning: unable to resolve reference to 'Base' for \ref command
/home/lahwaacz/stuff/test.cpp:21: warning: unable to resolve reference to 'member_1' for \ref command
```

It seems the relevant bits are that the class is a template and defined inside a namespace.

Full example attached: 
[bug.zip](https://github.com/doxygen/doxygen/files/9997336/bug.zip)

**Version**
* Doxygen version: 1.9.5
* Platform: Arch Linux (official [PKGBUILD](https://github.com/archlinux/svntogit-packages/blob/packages/doxygen/trunk/PKGBUILD))

As far as I can tell, version 1.9.3 (the previous one that was packaged for Arch Linux) did not produce warnings in such cases.

**Comment by albert-github** [comment]
The 1.9.3 (and 1.9.4) version gave the warnings about `Base` looks like here are some links missing as well.

**Comment by lahwaacz** [comment]
We've been hit by this in a project when the CI system has upgraded doxygen from 1.9.3 to 1.9.5 so there is some regression. Though I might have not isolated the dummy example perfectly to demonstrate it is actually a regression...

**Comment by isaac-pascual** [comment]
The same happen to our project when upgrade from 1.9.3 to 1.9.5. A lot of warnings but is like ref work well in generated doc.

**Comment by albert-github** [comment]
I looked a bit closer to the original problem and more precise to the line `Alias for the template parameter of \ref Base`.
As far as I can see in the `template< typename T >
class Base` part the check is against the context `foo:Base` to find `Base` and in the inherited part the check is against `Base< int >`. 
Probable cause might be that `getScopeDefs` in `util.cpp` doesn't look "deep enough" to find these inherited references or that nmaybe the context should resolve / replace `Base< int >` in an earlier stage with `foo::Base` (e.g. in `resolveRef` or in `resolveLink` or ...)

**Comment by doxygen** [comment]
@lahwaacz Please verify if the referenced commit fixes the problem for you.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
